common/BackArrow.js

```diff
@@ -0,0 +1,29 @@
+import EStyleSheet from "react-native-extended-stylesheet";
+import { Image, TouchableOpacity } from "react-native";
+
+
+
+
+
+
+function BackArrow(props) {
+
+    return (
+        <TouchableOpacity
+          style={styles.back}
+          onPress={() => props.navigation.navigate(props.location)}
```

Comment:
I looked into it and there's a hook called useNavigation(). It lets a non-navigation component use the navigation object. See following:
https://reactnavigation.org/docs/use-navigation

You can use this instead of passing navigation as a prop every time.